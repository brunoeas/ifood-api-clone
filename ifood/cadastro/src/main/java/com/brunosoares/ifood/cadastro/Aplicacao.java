package com.brunosoares.ifood.cadastro;

import javax.enterprise.context.ApplicationScoped;
import java.util.logging.Logger;

@ApplicationScoped
public class Aplicacao {

//    @Incoming("requests")
    public void process(String body) {
        Logger.getAnonymousLogger().info("mensagem recebida: {}" + body);
    }

}
