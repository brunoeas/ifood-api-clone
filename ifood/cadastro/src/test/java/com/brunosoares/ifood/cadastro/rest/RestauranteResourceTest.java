package com.brunosoares.ifood.cadastro.rest;

import com.brunosoares.ifood.cadastro.CadastroIfoodTestLifecycleManager;
import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroIfoodTestLifecycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testBuscarTodos() {
        final String res = given()
                .when()
                .get("/v1/restaurantes")
                .then()
                .statusCode(200)
                .extract()
                .asString();
        Approvals.verifyJson(res);
    }

}