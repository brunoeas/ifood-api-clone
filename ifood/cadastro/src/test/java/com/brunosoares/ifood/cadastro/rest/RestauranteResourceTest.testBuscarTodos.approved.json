[
  {
    "id": 123,
    "nome": "novo 1",
    "proprietario": "PROP"
  },
  {
    "id": 456,
    "nome": "novo 2",
    "proprietario": "PROP 2"
  }
]